src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -273,7 +269,9 @@ public virtual void Dispose()
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public LazyStringValue GetLazyStringForFieldWithCaching(StringSegment key)
-        {
+        {            
+            Debug.Assert(key.Length == key.String.Length, "If the segment do not comprise the whole string this method will fail.");
```

Comment:
Mhhh that makes sense I run tests on release mode. I added that as a safeguard, probably it is more complex, you can remove it without problem